### ISSUE TITLE
Variable font output: mixed up vertical metrics?

### ISSUE THREAD

**Issue by madig** [googlei18n/fontmake#492]
When I try to make a variable font from Cantarell, the default outlines seem to be vertically lower than the interpolated instances, at least in GNOME (no problem in Firefox).

1. Get https://gitlab.gnome.org/GNOME/cantarell-fonts
2. `fontmake -g Cantarell.glyphs -o variable`
3. Use e.g. `ftview 16 Cantarell-VF.ttf` to view and use `n`/`p` to switch between instances. The first is normal, the instances are shifted upwards it seems.

**Comment by madig** [comment]
Also, `head`'s yMin and yMax seems to be taken from the default instance instead of considering the slightly taller metrics of the black master.

**Comment by behdad** [comment]
varLib doesn't handle vertical glyph positions currently.  There's TODO items in there.  Perhaps you can fix that?

**Comment by behdad** [comment]
I can review.

**Comment by madig** [comment]
By vertical glyph positions you mean vertical metrics (hhea, typo, win)? I can have a look.

**Comment by behdad** [comment]
> By vertical glyph positions you mean vertical metrics (hhea, typo, win)? I can have a look.

No.

typo, win, vhea, etc are handled by MVAR, and those are implemented.

Per-glyph vertical metrics (vertical origin and advance) is not.  That's VVAR table, and vertical phantom points, vmtx.  Search for XXX in `_SetCoordinates` and `_GetCoordinates` in `varLib.__init__`.

**Comment by madig** [comment]
Uhm, I'm not using any of that or at least I think I don't. This is about
instances being shifted up vertically relative to the default outlines when
they shouldn't. It seems ok in Firefox. I can attach a TTF tomorrow morning.

**Comment by behdad** [comment]
Can you please rephrase the original report in more accurate terms?  Is this about font metrics being interpretted differently by FreeType for varfont instances compared to static font instances?

**Comment by madig** [comment]
Yes, the hhea, typo and win metrics seem to be interpreted differently for instances. The static master fonts and the default instance of the VF have the intended metrics, the instances' baselines are shifted upwards in relation. I can ask on freetype-devel if I'm missing something here.

**Comment by apodtele** [comment]
The following seems to be different: hhea does not seem to be used in VF. Compare and decide which one is correct.

https://git.savannah.gnu.org/cgit/freetype/freetype2.git/tree/src/truetype/ttgxvar.c#n1365
https://git.savannah.gnu.org/cgit/freetype/freetype2.git/tree/src/sfnt/sfobjs.c#n1670
